llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -4564,6 +4564,17 @@ static Instruction *foldICmpXorXX(ICmpInst &I, const SimplifyQuery &Q,
   return nullptr;
 }
 
```

Comment:
I have seen checks for `or` which is actually an `add` in other places. Should we make this into a centralized utility function?

Reply:
Great, looks like this comes up in other places too:  https://github.com/llvm/llvm-project/pull/72583  
I'll wait for that PR to land, so I can apply it here.

Reply:
Also see similar PR: https://github.com/llvm/llvm-project/pull/75899